### DISCUSSION TITLE
Do not run on_failure_callback in async

### DISCUSSION THREAD

**Discussion by matejpavlovic-maistra** [apache/airflow#19779] [Q&A]
If dag is defined as Task A >> Task B.
How to set that Task B waits for on_failure_callback of Task A to finishes before running.
Default DAG args will run Task B and on_failure_callback of Task A in parallel.

**Comment by potiuk** [comment]
I think this is against the idea of callback. Callbacks are asynchronous by nature. If you want to do something synchronously you should make sure the action that you run in callback is called during the "execute" method of the operator.
